### PR TITLE
Android 8 channel integration

### PR DESCRIPTION
**Pull request by da314pc**

### CODE REVIEW

File: lib/server/server.js [hunk 1]
```diff
@@ -2,17 +2,32 @@ Push.appCollection = new Mongo.Collection('_raix_push_app_tokens');
 Push.appCollection._ensureIndex({ userId: 1 });
 
 Push.addListener('token', function(currentToken, value) {
+  //console.log("currentToken",currentToken);
```
Comment (shivang007) [inline lib/server/server.js]: Remove all the commented consoles

File: lib/server/server.js [hunk 2]
```diff
@@ -80,20 +96,29 @@ Meteor.methods({
...
+      //     { token: { $exists: true } }  // Make sure token exists
+      //   ]
+      // });
+      if(doc.token)
+      { 
+        var removed = Push.appCollection.remove({ 
+          $and: [ { _id: { $ne: doc._id } },
+          { token: doc.token }, // Match token 
+          { appName: doc.appName }, // Match appName 
+          { token: { $exists: true } } // Make sure token exists
```
Comment (shivang007) [inline lib/server/server.js]: No need for checking $exists as you are already checking in if(doc.token).

Comment (da314pc) [reply lib/server/server.js]: I removed the commented out code

File: lib/server/server.js [hunk 3]
```diff
@@ -2,17 +2,32 @@ Push.appCollection = new Mongo.Collection('_raix_push_app_tokens');
...
+  //console.log("currentToken",currentToken);
+  //console.log("value",value);
   if (value) {
     // Update the token for app
     Push.appCollection.update({ token: currentToken }, { $set: { token: value } }, { multi: true });
-  } else if (value === null) {
+  } 
+  else if (value === null) {
     // Remove the token for app
+    var value;
```
[the remark this answers is not in the document]
Comment (da314pc) [reply lib/server/server.js]: I had a helper version in my version of this package but its not needed so I removed it.

**Comment by shivang007** [comment]
@da314pc I tried in multiple android mobiles, the notification does not have a default icon of my app. Instead it's just a green colored square!
Can you test if the same thing is happening with your build?

**Comment by shivang007** [comment]
> @da314pc I tried in multiple android mobiles, the notification does not have a default icon of my app. Instead it's just a green colored square!
> Can you test if the same thing is happening with your build?

Adding cordova-build-override with platform/android/res/drawable as @aboire suggested, solves this.

**Comment by shivang007** [comment]
> Can you post what you added so I can add that to the readme

Add your icon and logo in _cordova-build-override/platforms/android/res/drawable_
Add your sound file in _cordova-build-override/platforms/android/res/raw_
Like this:

**Comment by da314pc** [comment]
> > Can you post what you added so I can add that to the readme
> 
> Add your icon and logo in _cordova-build-override/platforms/android/res/drawable_
> Add your sound file in _cordova-build-override/platforms/android/res/raw_
> Like this:

I think we should use the default meteor way with the mobile config.js
App.icons({
    // Android
    'android_mdpi':     'resources/icons/48x.png',
    'android_hdpi':     'resources/icons/72x.png',
    'android_xhdpi':    'resources/icons/96x.png',
	  'android_xxhdpi': 	'resources/icons/144.png',
	  'android_xxxhdpi': 	'resources/icons/192x.png'
});

during the build

**Comment by shivang007** [comment]
@da314pc I am already doing that in my mobile-config.js.

**Comment by da314pc** [comment]
> @da314pc I am already doing that in my mobile-config.js.

If you were able to get the notifications working we can proceed with this pull request.
Anyone who needs icons , we can refer them to the meteor guide. to add them during the build with the mobile config settings or just copy and paste them in the build override folder.

**Comment by shivang007** [comment]
> > @da314pc I am already doing that in my mobile-config.js.
> 
> If you were able to get the notifications working we can proceed with this pull request.
> Anyone who needs icons , we can refer them to the meteor guide. to add them during the build with the mobile config settings or just copy and paste them in the build override folder.

Cool, push your latest code, resolve all the comments and we can proceed

**Comment by da314pc** [comment]
> > > @da314pc I am already doing that in my mobile-config.js.
> > 
> > 
> > If you were able to get the notifications working we can proceed with this pull request.
> > Anyone who needs icons , we can refer them to the meteor guide. to add them during the build with the mobile config settings or just copy and paste them in the build override folder.
> 
> Cool, push your latest code, resolve all the comments and we can proceed

I can't approve my own pull request. You guys have to review and approve.
@shivang007  @aboire

File: lib/server/server.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ Push.addListener('token', function(currentToken, value) {
 
 Meteor.methods({
   'raix:push-update': function(options) {
+    this.unblock();
```
Comment (shivang007) [inline lib/server/server.js]: Why do we need this?

Comment (da314pc) [reply lib/server/server.js]: @shivang007 good eye. Its been removed. 
On my branch in production, this method can be very slow sometimes. So I didn't want it slowing down other parts of my app.

Comment (da314pc) [reply lib/server/server.js]: dont worry its been removed from the pull request.

Comment (da314pc) [reply lib/server/server.js]: @shivang007 if you dont see anything else please approve. if not let me know what you see.

**Pull request merged**